packages/plugin-react/src/index.ts

```diff
@@ -108,36 +108,17 @@ export default function viteReact(opts: Options = {}): PluginOption[] {
         []
 
       if (/\.(mjs|[tj]sx?)$/.test(extension)) {
-        const plugins = [...userPlugins]
+        const isJSX = extension.endsWith('x')
+        const isNodeModules = id.includes('/node_modules/')
+        const isProjectFile =
+          !isNodeModules && (id[0] === '\0' || id.startsWith(projectRoot + '/'))
```

Comment:
If a virtual module wants to avoid being transformed by Babel, it can include `/node_modules/` in its module ID.